### PR TITLE
Avoid a crash in grpc_security_connector_shutdown.

### PR DESCRIPTION
**Pull request by nicolasnoble**

### CODE REVIEW

File: src/core/security/server_secure_chttp2.c [hunk 1]
```diff
@@ -148,7 +148,9 @@ static void destroy_done(grpc_exec_ctx *exec_ctx, void *statep, int success) {
     state->destroy_callback->cb(exec_ctx, state->destroy_callback->cb_arg,
                                 success);
   }
-  grpc_security_connector_shutdown(exec_ctx, state->sc);
+  if (state->sc) {
+    grpc_security_connector_shutdown(exec_ctx, state->sc);
+  }
   state_unref(state);
```
Comment (nicolasnoble) [inline src/core/security/server_secure_chttp2.c]: Turns out this one crashes too. We have a ref of 0 when we reach there. I don't have a good repro case at the moment unfortunately.

Comment (jboeuf) [inline src/core/security/server_secure_chttp2.c]: @yang-g did the work on the shutdown in #4175. Should he be the one reviewing this PR?

Comment (yang-g) [inline src/core/security/server_secure_chttp2.c]: Thanks for the headsup, I will look at why this can go wrong.

Comment (yang-g) [inline src/core/security/server_secure_chttp2.c]: #5032 should be the right fix.

**Pull request closed without merging**